src/v2/mod.rs

```diff
@@ -2102,6 +2102,120 @@ impl Client {
         });
         Ok(FinalizedBlocksStream { handle, receiver })
     }
+
+    /// Find a block in which the instance was created, if it exists and is
+    /// finalized. The return value is a triple of the absolute block height and
+    /// the corresponding block hash, and the instance information at the
+    /// end of that block. The block is the first block in which the instance
+    /// appears.
+    ///
+    /// Note that this is not necessarily the initial state of the instance
+    /// since there can be transactions updating the instance in the same block
+    /// as the initialization transaction.
+    ///
+    /// Optional bounds can be provided, and the search will only
+    /// consider blocks in that range, inclusive. If the lower bound is not
+    /// provided it defaults to 0, if the upper bound is not provided it
+    /// defaults to the last finalized block at the time of the call.
+    ///
+    /// The return value is a pair of the absolute block height, and the hash,
+    /// of the block in which the instance was created. If the instance
+    /// cannot be found [`QueryError::NotFound`] is returned.
```

Comment:
I feel like the `QueryError::NotFound` is describing something different from this case, since it is not the same as the rpc query failing to find a resource.

Reply:
I don't agree with this. The resource is not found in this case either.

I'm not sure why we need to distinguish "the node says the contract does not exist in a specific block" and "we could not find it".

Reply:
I guess it is fine in this particular case.